internal/cmd/beta/security-group/update/update.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/stackitcloud/stackit-cli/internal/pkg/flags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/projectname"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/iaas/client"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/utils"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	Labels          *map[string]string
+	Description     *string
+	Name            *string
+	SecurityGroupId string
+}
+
+const groupNameArg = "GROUP_ID"
+
+const (
+	nameArg        = "name"
+	descriptionArg = "description"
+	labelsArg      = "labels"
+)
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Updates a security group",
+		Long:  "Updates a named security group",
+		Args:  args.SingleArg(groupNameArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(`Update the name of a group`, `$ stackit beta security-group update 541d122f-0a5f-4bb0-94b9-b1ccbd7ba776 --name my-new-name`),
+			examples.NewExample(`Update the labels of a group`, `$ stackit beta security-group update 541d122f-0a5f-4bb0-94b9-b1ccbd7ba776 --labels label1=value1,label2=value2`),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			projectLabel, err := projectname.GetProjectName(ctx, p, cmd)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get project name: %v", err)
+				projectLabel = model.ProjectId
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+
+			_, err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("^date security group: %w", err)
+			}
+			p.Info("Updated security group \"%v\" for %q\n", model.Name, projectLabel)
+
+			return nil
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().String(nameArg, "", "The name of the security group.")
+	cmd.Flags().String(descriptionArg, "", "An optional description of the security group.")
+	cmd.Flags().StringToString(labelsArg, nil, "Labels are key-value string pairs which can be attached to a network-interface. E.g. '--labels key1=value1,key2=value2,...'")
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, cliArgs []string) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	if err := cmd.ValidateArgs(cliArgs); err != nil {
+		return nil, &errors.ArgValidationError{
+			Arg:     groupNameArg,
+			Details: fmt.Sprintf("argument validation failed: %v", err),
+		}
+	}
```

Comment:
This is not necessary here, as the arg validation function is called by default by Cobra. See e.g. other commands like https://github.com/stackitcloud/stackit-cli/blob/c715b0ac904913e15db4550b8710f2910b6e39a3/internal/cmd/beta/server/update/update.go#L42